integration_test/test_xgboost_experiment.py

```diff
@@ -104,6 +104,20 @@ def test_experiment(self, task):
     self._verify_metrics_for_all_runs(sigopt_suggested_runs, experiment.metrics)
     experiment.archive()
 
+  def test_early_stopping(self):
+    task = 'binary'
+    self._form_random_experiment_config(task)
+    self.experiment_params['early_stopping_rounds'] = 1
+    experiment = sigopt.xgboost.experiment(**self.experiment_params)
+    assert experiment.is_finished()
+    sigopt_suggested_runs = list(experiment.get_runs())
+    for ssr in sigopt_suggested_runs:
+      assert ssr.assignments['early_stopping_rounds']
+      for param in experiment.parameters:
+        if param.name == 'num_boost_round':
+          assert ssr.values['num_boost_round_before_stopping']
+          assert ssr.assignments['num_boost_round'] >= ssr.assignments['num_boost_round_before_stopping']
```

Comment:
^ If you test never caught this, that means this test is not very useful. My guess is that most of these runs would not be early stopped (because `num_boost_round` is very small already).

It's sufficient to check whether `early_stopping_round` is passed to the run level.